server/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -151,9 +155,6 @@ public static void parseSearchRequest(SearchRequest searchRequest, RestRequest r
         }
 
         String types = request.param("type");
-        if (types != null) {
-            deprecationLogger.deprecated("The {index}/{type}/_search endpoint is deprecated, use {index}/_search instead");
-        }
```

Comment:
Does this need keeping? It warns when invoking URLs of this form:

     GET my_index/_search?type=my_type

Reply:
Maybe not for this PR but while we're itemising the ways clients can filter by type this currently also raises no warnings:

    GET my_index/_search?q=_type:my_type

or other forms of query:

	GET my_index/_search
	{
	  "query":{
		"match":{
		  "_type":"my_type"
		}
	  }
	}

Reply:
Thanks @markharwood! I hadn't considered query parameters at all in this PR, will fix that. I added a note about queries that use `_type` as a field name to #35190.